jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentManager.java

```diff
@@ -181,7 +180,7 @@ public void addAppProvider(AppProvider provider)
         if (isRunning())
             throw new IllegalStateException();
         _providers.add(provider);
-        addBean(provider);        
+        addManaged(provider);
```

Comment:
I think we want to leave `addBean()` here.
If user code passes in an already started `AppProvider` we should not manage it.

Reply:
Typically, the `addAppProvider` occurs in the `jetty-deploy.xml`, which makes it difficult to setup managed (or not).

Reply:
It's a `public` method people can call so you can get a `AppProvider` that is already started and used somewhere else and you don't want to manage.
In the normal case you mention, `addBean()` and `addManaged()` will behave the same. The former also covers my example use case, so I don't see why change it.